Core/ALife.Core/WorldObjects/Agents/Senses/ProximityCluster.cs

```diff
@@ -21,29 +20,29 @@ public override IShape Shape
             }
         }
 
-        public ProximityCluster(WorldObject parent, string name, EvoNumber radius)
+        public ProximityCluster(WorldObject parent, string name, IEvoNumber radius)
             : base(parent, name)
         {
             evoRadius = radius;
 
-            myShape = new ChildCircle(parent.Shape, new Angle(0), 0, (float)radius.StartValue);
+            myShape = new ChildCircle(parent.Shape, new Angle(0), 0, (float)radius.OriginalValue);
 
             SubInputs.Add(new AnyInput(name + ".SomethingClose"));
         }
-        public ProximityCluster(WorldObject parent, string name, EvoNumber radius, Color newColor)
+        public ProximityCluster(WorldObject parent, string name, IEvoNumber radius, Colour newColor)
             : this(parent, name, radius)
         {
-            myShape.Color = newColor;
+            myShape.Colour = newColor;
         }
 
         public override SenseCluster CloneSense(WorldObject newParent)
         {
-            return new ProximityCluster(newParent, Name, evoRadius.Evolve(), myShape.Color.Clone());
+            return new ProximityCluster(newParent, Name, evoRadius.Evolve(Planet.World.NumberGen), (Colour)myShape.Colour.Clone());
```

Comment:
I don't want to pass in the NumberGen here. That implies or allows for multiple number generators to be used, which we definitely do not want. There is a global number generator, and everyone uses it.

Reply:
So. One of the big things I am trying to work away from in my V2 proposal is having a singleton version of the entire program (which was causing issues with some of the Avalonia work I was doing). With how things work atm, you are right, you could theoretically spin off a new random gen here and pass it across. In my V2 proposal, we'd technically be passing around a Simulation instance which would be the equivalent of the Planet.Workd thing (and would house the random gen for that simulation instance). Since we don't have that yet, this is the next best thing to simulate how that will work